### PR TITLE
refactor(customjs2): processor name is unique alongside id

### PR DESCRIPTION
**Pull request by syncush**

### CODE REVIEW

File: src/processors/models/database/cassandra/cassandraConnector.js [hunk 1]
```diff
@@ -32,19 +47,31 @@ async function getAllProcessors(from, limit) {
...
+        let mappingParams = [processor.name];
+        return Promise.all([executeQuery(DELETE_PROCESSOR, params, queryOptions), executeQuery(DELETE_PROCESSOR_MAPPING, mappingParams, queryOptions)]);
+    }
 }
 
 async function insertProcessor(processorId, processorInfo) {
     let params = [processorId, processorInfo.name, processorInfo.description, processorInfo.javascript, Date.now(), Date.now()];
-    const processor = await executeQuery(INSERT_PROCESSOR, params, queryOptions);
+    let mappingParams = [processorInfo.name, processorId];
+    const [processor] = await Promise.all([executeQuery(INSERT_PROCESSOR, params, queryOptions), executeQuery(INSERT_PROCESSOR_MAPPING, mappingParams, queryOptions)]);
```
Comment (NivLipetz) [inline src/processors/models/database/cassandra/cassandraConnector.js]:
```suggestion
    const [processor] = await Promise.all([
          executeQuery(INSERT_PROCESSOR, params, queryOptions), 
          executeQuery(INSERT_PROCESSOR_MAPPING, mappingParams, queryOptions)
    ]);
```

File: src/processors/models/database/cassandra/cassandraConnector.js [hunk 2]
```diff
@@ -32,19 +47,31 @@ async function getAllProcessors(from, limit) {
...
 }
 
-function deleteProcessor(processorId) {
+async function deleteProcessor(processorId) {
     let params = [processorId];
-    return executeQuery(DELETE_PROCESSOR, params, queryOptions);
+    let processor = await getProcessorById(processorId);
+    if (processor) {
+        let mappingParams = [processor.name];
+        return Promise.all([executeQuery(DELETE_PROCESSOR, params, queryOptions), executeQuery(DELETE_PROCESSOR_MAPPING, mappingParams, queryOptions)]);
```
Comment (NivLipetz) [inline src/processors/models/database/cassandra/cassandraConnector.js]:
```suggestion
        return Promise.all([
            executeQuery(DELETE_PROCESSOR, params, queryOptions), 
            executeQuery(DELETE_PROCESSOR_MAPPING, mappingParams, queryOptions)
        ]);
```

File: src/processors/models/database/cassandra/cassandraConnector.js [hunk 3]
```diff
@@ -5,17 +5,32 @@ let client;
...
 const INSERT_PROCESSOR = 'INSERT INTO processors(id, name, description, javascript, created_at, updated_at) values(?,?,?,?,?,?)';
 const GET_ALL_PROCESSORS = 'SELECT * FROM processors';
-const GET_PROCESSOR = 'SELECT * FROM processors WHERE id=?';
+const GET_PROCESSOR_BY_ID = 'SELECT * FROM processors WHERE id=?';
 const DELETE_PROCESSOR = 'DELETE FROM processors WHERE id=?';
 const UPDATE_PROCESSOR = 'UPDATE processors SET name=?, description=?, javascript=?, updated_at=? WHERE id=? AND created_at=? IF EXISTS';
 
+const INSERT_PROCESSOR_MAPPING = 'INSERT INTO processors_mapping(name, id) VALUES(?, ?)';
+const DELETE_PROCESSOR_MAPPING = 'DELETE FROM processors_mapping WHERE name=?';
+const GET_PROCESSOR_MAPPING = 'SELECT * FROM processors_mapping WHERE name=?';
```
Comment (NivLipetz) [inline src/processors/models/database/cassandra/cassandraConnector.js]: missing `UPDATE_PROCESSOR_MAPPING` that should be used under function `updateProcessor` if the name is updated

File: src/processors/models/database/cassandra/cassandraConnector.js [hunk 4]
```diff
@@ -32,19 +47,31 @@ async function getAllProcessors(from, limit) {
...
-    const processor = await executeQuery(GET_PROCESSOR, [processorId], queryOptions);
+async function getProcessorByName(processorName) {
+    const [processorMapping] = await executeQuery(GET_PROCESSOR_MAPPING, [processorName], queryOptions);
+    if (processorMapping) {
+        return getProcessorById(processorMapping.id);
+    }
+}
+
+async function getProcessorById(processorId) {
+    const processor = await executeQuery(GET_PROCESSOR_BY_ID, [processorId], queryOptions);
```
Comment (NivLipetz) [inline src/processors/models/database/cassandra/cassandraConnector.js]:
```suggestion
    const [processor] = await executeQuery(GET_PROCESSOR_BY_ID, [processorId], queryOptions);
```

File: src/processors/models/processorsManager.js [hunk 5]
```diff
@@ -9,6 +9,10 @@ const logger = require('../../common/logger'),
 
 module.exports.createProcessor = async function (processor) {
     let processorId = uuid.v4();
+    const processorWithTheSameName = await databaseConnector.getProcessorByName(processor.name);
```
Comment (NivLipetz) [inline src/processors/models/processorsManager.js]: check/throw error first in the function before the `let processorId = uuid.v4();`

Comment (syncush) [reply src/processors/models/processorsManager.js]: done

File: src/processors/models/database/cassandra/cassandraConnector.js (hunk 3 above)
Comment (syncush) [reply src/processors/models/database/cassandra/cassandraConnector.js]: Cassandra doesn't support updating a record's primary key, therefore I need to delete and insert a new key-value record

File: src/processors/models/database/cassandra/cassandraConnector.js (hunk 2 above)
Comment (syncush) [reply src/processors/models/database/cassandra/cassandraConnector.js]: done

File: src/processors/models/database/cassandra/cassandraConnector.js (hunk 1 above)
Comment (syncush) [reply src/processors/models/database/cassandra/cassandraConnector.js]: done

**Pull request merged**